cmd/config/internal/commands/cmddeletesetter_test.go

```diff
@@ -0,0 +1,304 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package commands_test
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/cmd/config/ext"
+	"sigs.k8s.io/kustomize/cmd/config/internal/commands"
+	"sigs.k8s.io/kustomize/kyaml/openapi"
+)
+
+func TestDeleteSetterCommand(t *testing.T) {
+	var tests = []struct {
+		name              string
+		input             string
+		args              []string
+		schema            string
+		out               string
+		inputOpenAPI      string
+		expectedOpenAPI   string
+		expectedResources string
+		err               string
+	}{
+		{
+			name: "delete replicas",
+			args: []string{"replicas", "hello world"},
+			input: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 3 # {"$openapi" : "replicas"}}
+ `,
+			inputOpenAPI: `
+apiVersion: v1alpha1
+kind: Example
+openAPI:
+  definitions:
+    io.k8s.cli.setters.replicas:
+      description: hello world
+      x-k8s-cli:
+        setter:
+          name: replicas
+          value: "3"
+          setBy: me
+`,
+			expectedOpenAPI: `
+apiVersion: v1alpha1
+kind: Example
+openAPI:
+  definitions: {}
```

Comment:
If there are no definitions then it's better to remove this section completely. This would create json definitions upon creating a new setter. Try creating a setter with this file and you can see the difference.

Reply:
I see, removed, also removed the empty openApi